### ISSUE TITLE
What would it take to be production-ready?

### ISSUE THREAD

**Issue by chrisvfritz** [gaearon/react-transform-boilerplate#84]
The docs make it very clear that this is not production-ready, but is the only thing keeping it from being so the hardcoded reference to `/static/bundle.js` in index.html? If so, I have an idea on getting it there without introducing extra complexity and would be happy to submit a PR.

**Comment by gaearon** [comment]
Maybe we can have `index.dev.html` and `index.prod.html`. The first one would be used in dev, and the second one would be copied to `dist` on build. What do you think?

**Comment by thejameskyle** [comment]
Well, do you want this to be a universal boilerplate as well?

**Comment by gaearon** [comment]
If we make it universal, I'd want to have two root folders: `client-only` and `universal`. Many people don't use Node in production, and will be confused by which parts of the boilerplate are needed just for the universal part.

**Comment by thejameskyle** [comment]
Ah ok

**Comment by chrisvfritz** [comment]
I think it'd be best to keep this client-only, to avoid the added complexity and maintenance cost. There are other boilerplate projects for universal apps and the choice in backend has nothing to do with the dev environment this project is meant to demo.

@gaearon I may be missing something, but these two simple changes seem to work for me:

- Change the `script` tag's `src` in `index.html` to `"/dist/bundle.js"`
- Update the `publicPath` in `webpack.config.dev.js` to `'/dist/'`

To confirm it worked in both development and production, I ran:

1. `npm run start` and confirmed hot-loading was working
2. `npm run build` and deployed to a web server, confirming that worked
3. `npm run start` again just to confirm that existing files in `/dist` didn't interfere with the hot-loading

Is it possible that this is sufficient? I admit to being no more than a passable Webpack user, so it's very possible there's something I'm missing. Please don't hesitate to let me know what it is. :smiley:

**Comment by chrisvfritz** [comment]
Can anyone else confirm that this works for them? Should I submit a PR?

**Comment by dsifford** [comment]
@chrisvfritz Saw your suggestion last night... Still working on stringing things together in my application to test a real-world implementation. Having said that, it does work as you stated for me using the boilerplate. :+1: 

(Note: Like you, I am no expert either when it comes to webpack, so I could also be overlooking something).

**Comment by EmperorEarth** [comment]
Alternatively, we can have the master (client-only) branch and a universal branch?

**Comment by chrisvfritz** [comment]
This issue isn't about incorporating a backend (universal or not) - it sounds like that's out of scope for this project, which is focused on the development environment. But as it currently stands, this boilerplate isn't even ready for a production _static_ site. #93 fixes this.

**Comment by x4080** [comment]
i wondered about this too, so its ready for production?

**Comment by chrisvfritz** [comment]
@x4080 not until #93 is merged.

**Comment by x4080** [comment]
Ok @chrisvfritz

**Comment by gaearon** [comment]
Should work for building a static site now.
